love/components/search-location.tsx

```diff
@@ -0,0 +1,154 @@
+// src/components/SearchBar.tsx
+import clsx from 'clsx'
+import { Lover } from 'love/hooks/use-lover'
+import { useEffect, useRef, useState } from 'react'
+import { Col } from 'web/components/layout/col'
+import { Row } from 'web/components/layout/row'
+import { Input } from 'web/components/widgets/input'
+import { searchLocation } from 'web/lib/firebase/api'
+import { Row as rowFor } from 'common/supabase/utils'
+import { LoadingIndicator } from 'web/components/widgets/loading-indicator'
+
+export type City = {
+  city: string
+  regionCode: string
+  country: string
+  latitude: number
+  longitude: number
+}
+
+function loverToCity(lover: rowFor<'lovers'>) {
+  return {
+    city: lover.city,
+    regionCode: lover.region_code,
+    country: lover.country,
+    latitude: lover.latitude,
+    longitude: lover.longitude,
+  } as City
+}
+
+export default function CitySearchBox(props: {
+  onCitySelected: (city: City | undefined) => void
+  lover: rowFor<'lovers'>
+}) {
+  const { onCitySelected, lover } = props
+  const [query, setQuery] = useState('')
+  const [cities, setCities] = useState<City[]>([])
+  const [loading, setLoading] = useState(false)
+  const [dropdownVisible, setDropdownVisible] = useState(false)
+  const dropdownRef = useRef<HTMLUListElement>(null)
+
+  useEffect(() => {
+    if (query.length < 2) {
+      setCities([])
+    }
+  }, [query])
+
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
Could keep a `useRef` with the query count, and only set state if it's the latest query count.